unityfiles/Assets/Scripts/Game/Menu/PauseMenu.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PauseMenu : MonoBehaviour {
+    [SerializeField]
+    private GameObject pauseMenuCanvas;
+
+    [SerializeField]
+    private GameObject confirmDialog;
+
+    private Boolean isPaused = false;
+
+    private Button continueButton;
+    private static readonly int PAUSE = 0;
+    private static readonly float PLAY = 1;
+
+    private void Awake() {
+        CheckForMissingComponents();
+        
+        DeactivateMenu(); 
+    }
+
+    private void Start() {
+        SubscribeToEvents();
+    }
+
+    // Update is called once per frame
+    void Update() {
+        if (Input.GetButtonDown("Cancel")) {
+            if (confirmDialog.activeSelf) {
+                DeactivateConfirmDialog();
+            } else {
+                TogglePause();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Toggles the pause state.
+    /// Shows pause menu if paused
+    /// </summary>
+    private void TogglePause() {
+        isPaused = !isPaused;
+
+        if (isPaused) {
+            PauseGame();
+        } else {
+            ResumeGame();
+        }
+    }
+
+    private void OnDestroy() {
+        UnsubscribeFromEvents();
+    }
+
+    private void UnsubscribeFromEvents() {
+        LevelManager.LevelStateChangeEvent -= ChangeGameState();
+    }
+
+    private void SubscribeToEvents() {
+        LevelManager.LevelStateChangeEvent += ChangeGameState();
+    }
+
+    private EventHandler<LevelStateChangeEventArgs> ChangeGameState() {
+        return (sender, args) => {
+            if (args.NewState == LEVEL_STATE.PAUSE) {
+                PauseGame();
+            }
+
+            if (args.NewState == LEVEL_STATE.RESUME) {
+                if (confirmDialog.activeSelf) {
+                    DeactivateConfirmDialog();
+                } else {
+                    ResumeGame();
+                }
+            }
+
+            if (args.NewState == LEVEL_STATE.CONFIRM) {
+                ActivateConfirmDialog();
+            }
+        };
+    }
```

Comment:
Du skal ikke trenge å lytte på disse eventene. Om noen sitter LevelState til pause ved f.eks, inventory eller ved GameOver å vise stats eller noe slikt før scene bytte. Vil det gjøre at din meny spretter opp.

GUI knappene dine burde være ansvarlig for å trigge den logikken her. Du kan tappe deg inn på en knapp click event istedefor.

Eksempel
```c#
[SerializeField]
private Button resumeButton;

    private void Awake() {
        CheckForMissingComponents();
        DeactivateMenu(); 
        resumeButton.onClick.AddListener(ResumeGame);
    }


    private void OnDestroy() {
        resumeButton.onClick.RemoveListener(ResumeGame);
    }

Reply:
Done